src/tests/ftest/nvme/nvme_health.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+import os
+from nvme_utils import ServerFillUp, get_device_ids
+from test_utils_pool import TestPool
+from dmg_utils import DmgCommand
+from command_utils_base import CommandFailure
+
+class NvmeHealth(ServerFillUp):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: To validate NVMe health test cases
+    :avocado: recursive
+    """
+    def test_monitor_for_large_pools(self):
+        """Jira ID: DAOS-4722.
+
+        Test Description: Test Health monitor for large number of pools.
+        Use Case: This tests will create the 40 number of pools and verify the
+                  dmg list-pools, device-health and nvme-health works for all
+                  pools.
+
+        :avocado: tags=all,hw,medium,nvme,ib2,full_regression
+        :avocado: tags=nvme_health
+        """
+        # pylint: disable=attribute-defined-outside-init
+        # pylint: disable=too-many-branches
+        no_of_pools = self.params.get("number_of_pools", '/run/pool/*')
+        #Stop the servers to run SPDK too to get the server capacity
+        self.stop_servers()
+        storage = self.get_server_capacity()
+        self.start_servers()
+
+        single_pool_nvme_size = int((storage * 0.80)/no_of_pools)
+
+        pools = []
```

Comment:
Optional: If you will remove this line and mention self.pool = [] in your init function, you will not have to destroy pool in the end.

Reply:
Well I am not sure if multiple pool can be deleted by default. It will only remove the last pool created as part of pool class.
So I had to delete it each pool.

Reply:
If self.pool = [] is initialized then all pools should be deleted by default.